src/Mod/Draft/draftutils/init_draft_statusbar.py

```diff
@@ -0,0 +1,144 @@
+# *                                                                         *
+# *   This file is part of the FreeCAD CAx development system.              *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   FreeCAD is distributed in the hope that it will be useful,            *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with FreeCAD; if not, write to the Free Software        *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+import FreeCAD as App
+import FreeCADGui as Gui
+from PySide import QtGui
+from PySide.QtCore import QT_TRANSLATE_NOOP
+
+
+
+def set_draft_statusbar(show=True):
+    
+    draft_scales =  [QT_TRANSLATE_NOOP("draft","1:1000"),
+                     QT_TRANSLATE_NOOP("draft","1:500"),
+                     QT_TRANSLATE_NOOP("draft","1:250"),
+                     QT_TRANSLATE_NOOP("draft","1:200"),
+                     QT_TRANSLATE_NOOP("draft","1:100"),
+                     QT_TRANSLATE_NOOP("draft","1:50"),
+                     QT_TRANSLATE_NOOP("draft","1:25"),
+                     QT_TRANSLATE_NOOP("draft","1:20"),
+                     QT_TRANSLATE_NOOP("draft","1:10"),
+                     QT_TRANSLATE_NOOP("draft","1:5"),
+                     QT_TRANSLATE_NOOP("draft","1:2"),
+                     QT_TRANSLATE_NOOP("draft","1:1"),
+                     QT_TRANSLATE_NOOP("draft","2:1"),
+                     QT_TRANSLATE_NOOP("draft","5:1"),
+                     QT_TRANSLATE_NOOP("draft","10:1"),
+                     QT_TRANSLATE_NOOP("draft","20:1"),                 
+                     QT_TRANSLATE_NOOP("draft","custom"),                 
+                    ]
+    
+    param = App.ParamGet("User parameter:BaseApp/Preferences/Mod/Draft")
+    draft_annotation_scale = param.GetFloat("DraftAnnotationScale", 1.0)
+    
+    def set_scale(action):
```

Comment:
Here you are defining a function within a function.

I suggest you take this piece of code and move it outside. As far as I can tell, it's just a callback for some widget.

Using nested functions is, of course, possible, however, this type of coding is used in cases where the outer function needs to generate an internal function on the fly every time it's called. This is an advanced coding style that can be used for enhanced security as we "hide" the inner function. But I don't recommend it in this case, as it just makes the inner code more indented, and harder to read.

Since we are importing only the outer function, `set_draft_statusbar`, the inner function could be named `_set_scale`. The initial underscore `_` is a convention in Python to indicate that this function will not be imported normally, so it is meant only for internal use in the module.

See [Python Inner Functions—What Are They Good For?](https://realpython.com/inner-functions-what-are-they-good-for/)